### PR TITLE
runner: Implement validator support

### PR DESCRIPTION
**Pull request by Xuanwo**
This PR is a PoC for #13 to get a quick preview of my idea.

Welcome any comments.

### CODE REVIEW

**Comment by Xuanwo** [comment]
ping @skyzh to take a look.

**Comment by Xuanwo** [comment]
Please let me know if we need an RFC for this feature.

File: src/runner.rs [hunk 1]
```diff
@@ -82,16 +82,24 @@ impl TestErrorKind {
     }
 }
 
+type Validator = fn(&Vec<String>, &Vec<String>) -> bool;
```
Comment (skyzh) [inline src/runner.rs:85]: pub type? Please also add docstring for this type.

**Review by skyzh — COMMENTED** [review]
Rest LGTM, thanks!

By the way, can we go a step further? e.g., add a `Result` associated type for `DB`, so that we can pass any type apart from `String` as query result. (Not needed in this PR)

Also, I would recommend adding an example for validator in `examples` folder.

**Comment by Xuanwo** [comment]
> By the way, can we go a step further? e.g., add a `Result` associated type for `DB`, so that we can pass any type apart from `String` as query result. (Not needed in this PR)

Oh, so nice. Let's do it.

File: src/runner.rs (hunk 1 above)
Comment (Xuanwo) [reply src/runner.rs]: addressed.

**Comment by Xuanwo** [comment]
> By the way, can we go a step further? e.g., add a `Result` associated type for `DB`, so that we can pass any type apart from `String` as query result. (Not needed in this PR)

Created a new issue #16 to track it.

**Comment by Xuanwo** [comment]
> Also, I would recommend adding an example for validator in `examples` folder.

Added. Maybe we should move some of them into `tests` instead.

**Review by skyzh — APPROVED** [review]

**Pull request merged**